File: flask-backend/controllers/google_index_controller.py
import requests
from bs4 import BeautifulSoup

def get_google_index(url):
    try:
        search_url = f'https://www.google.com/search?q=site:{url}'
        headers = {'User-Agent': 'Mozilla/5.0'}
        
        response = requests.get(search_url, headers=headers)
        soup = BeautifulSoup(response.text, 'html.parser')
        
        # Check if any results found
        results = soup.find('div', {'id': 'result-stats'})
        is_indexed = results is not None
        
        return {
            'url': url,
            'is_indexed': is_indexed,
            'total_results': results.text if results else '0'
        }
    except Exception as e:
        print(f'Google Index Error: {str(e)}')
        return {'url': url, 'is_indexed': False, 'total_results': '0'}

File: nodejs-backend/googleIndexController.js
async function getGoogleIndex(url) {
    try {
        const apiKey = process.env.GOOGLE_API_KEY;
        const cx = process.env.GOOGLE_SEARCH_ENGINE_ID;
        
        const searchUrl = `https://www.googleapis.com/customsearch/v1?key=${apiKey}&cx=${cx}&q=site:${encodeURIComponent(url)}`;
        
        const response = await fetch(searchUrl);
        const data = await response.json();
        
        const isIndexed = data.searchInformation?.totalResults > 0;
        console.log('Search results:', {
            url,
            totalResults: data.searchInformation?.totalResults,
            isIndexed
        });
        
        return { isIndexed: isIndexed ? 1 : 0 };
    } catch (error) {
        console.error('Google Index Error:', error);
        return { isIndexed: 0 };
    }
};

export default getGoogleIndex;